src/schema.c

```diff
@@ -56,30 +56,28 @@ ldap_schema_new(TALLOC_CTX *ctx)
         return NULL;
     }
 
-    result->attribute_types = talloc_zero_array(ctx, LDAPAttributeType*, 1024);
+    result->object_classes = g_hash_table_new(g_str_hash, g_str_equal);
 
-    if (!result->attribute_types)
+    if (!result->object_classes)
     {
-        ld_error("Unable allocate attribute types in schema: %d ", result);
+        talloc_free(result);
+
+        ld_error("ldap_schema_new - out of memory - unable to create object classes in schema!\n");
 
         return NULL;
     }
 
-    result->attribute_types_capacity = 1024;
-    result->attribute_types_size = 0;
-
-    result->object_classes = talloc_zero_array(ctx, LDAPObjectClass*, 1024);
+    result->attribute_types = g_hash_table_new(g_str_hash, g_str_equal);
```

Comment:
Why have you change order of the initialization of `attribute_types` and `object_classes`?

Reply:
Please restore original order at first initialize `attribute_types` and than `object_classes`.